Hubert drewlo - task 3

Drewienkopl opened the pull request:
I have made fixes according to what was pointed out and everything should now be good.

<img width="1921" height="1042" alt="swaggerWorks" src="https://github.com/user-attachments/assets/5dc59dfd-2eb8-41c2-a11b-1e777f0ed7d2" />
<img width="1916" height="1041" alt="swaggerWorks2" src="https://github.com/user-attachments/assets/60ae0270-7e1f-4fe9-bd6f-13ac706b119a" />
<img width="1923" height="1040" alt="swaggerWorks3" src="https://github.com/user-attachments/assets/61b84216-b29d-42c3-904c-e68d26678904" />
<img width="1914" height="1037" alt="swaggerWorks4" src="https://github.com/user-attachments/assets/6dd49e1f-306c-4dc6-badd-c7e9e6a7ca45" />
<img width="1919" height="1044" alt="swaggerWorks5" src="https://github.com/user-attachments/assets/bdeaf7f5-a602-482c-a05f-d50464b71dc7" />
<img width="1924" height="1041" alt="swaggerWorks6" src="https://github.com/user-attachments/assets/f0f50b7b-e396-4cd5-b806-e18622dbff4d" />
<img width="1920" height="1041" alt="swaggerWorks7" src="https://github.com/user-attachments/assets/d63a005c-4937-4e96-b486-f1838a75fa99" />

Drewienkopl committed:
feat: task 3 finished

Drewienkopl committed:
Merge branch 'hubert-drewlo' into hubert-drewlo-3

Drewienkopl committed:
fix: trying to fix errors in lint

Drewienkopl committed:
fix: deleting specs

Drewienkopl committed:
Merge branch 'hubert-drewlo' into hubert-drewlo-3

Drewienkopl committed:
fix: turning off e2e tests

Drewienkopl committed:
fix: fixing errors

Drewienkopl committed:
Merge branch 'hubert-drewlo-3' of https://github.com/Solvro/backend-wakacyjne-wyzwanie-2025-zadania into hubert-drewlo-3

Drewienkopl committed:
fix: some warnings

Drewienkopl committed:
fix: prettier

Drewienkopl committed:
fix: prettier

Leadman5555 commented:
You need to finish part 2 first before starting part 3.

Leadman5555 commented:
> You need to finish part 2 first before starting part 3.

That is: #102

Leadman5555 commented on src/expense/dto/create-expense.dto.ts:
```diff
@@ -0,0 +1,39 @@
+
+  @ApiProperty({ required: false, example: "Train tickets" })
+  @IsOptional()
+  @ApiPropertyOptional()
```
Stacking both `@ApiPropertyOptional` and `@ApiProperty` makes the autogenerated definition behave in a weird way. Choose only one of them. Also, your linter should have warned you about it

Leadman5555 commented on src/expense/expense.controller.ts:
```diff
@@ -0,0 +1,62 @@
+    return this.service.create(dto);
+  }
+
+  @Patch(":id")
```
You are missing the operation tags for Swagger. Add them to **every** controller endpoint. Include the following two fields:
- summary
- return value schema

Leadman5555 commented on src/expense/expense.service.ts:
```diff
@@ -0,0 +1,76 @@
+  async create(dto: CreateExpenseDto) {
+    return this.prisma.expense.create({
+      data: {
+        participantId: dto.participantId,
```
What if the participant of the given Id does not exist? We need to check that first

Leadman5555 commented on src/expense/expense.service.ts:
```diff
@@ -0,0 +1,76 @@
+        amount: dto.amount,
+        category: dto.category,
+        note: dto.note,
+        participantId: dto.participantId,
```
Same thing here

Leadman5555 commented on src/expense/expense.controller.ts:
```diff
@@ -0,0 +1,62 @@
+  }
+
+  @Patch(":id")
+  @ApiOkResponse({ description: "Updated expense" })
```
Is 200 the only possible return code you can get here? After changing the service methods as requested, think what other codes should go here. Of course, omit the 500

Leadman5555 commented on src/participant/dto/create-participant.dto.ts:
```diff
@@ -0,0 +1,37 @@
+
+import { ApiProperty, ApiPropertyOptional } from "@nestjs/swagger";
+
+export class CreateParticipantDto {
```
Same here. Fix the ApiProperty decorators

Leadman5555 commented on src/participant/dto/create-participant.dto.ts:
```diff
@@ -0,0 +1,37 @@
+export class CreateParticipantDto {
+  @ApiProperty({ example: "Marek" })
+  @IsString()
+  name!: string;
```
Quick question, what is the exclamation mark (!) doing here? First of all, it might be treated like a normal character here. And even if it were not, it should not be here.
Overall, the only place you should use the non-null operator is when you are 100% sure that the value is not nullish, you have checked it, but the linter still does not understand it - such situations happen. And even then, you should think twice whether the linter really is wrong.

Leadman5555 commented on src/participant/participant.controller.ts:
```diff
@@ -0,0 +1,62 @@
+
+@ApiTags("participants")
+@Controller("participants")
+export class ParticipantController {
```
Same things to fix as in the previous controller

Leadman5555 commented on src/trip/trip.controller.ts:
```diff
@@ -0,0 +1,30 @@
+  @ApiOperation({ summary: "Create a new trip" })
+  @ApiOkResponse({ description: "The created trip" })
+  async createTrip(
+    @Body() body: { name: string; startDate: string; budget?: number },
```
Replace this with a proper DTO.

Leadman5555 reviewed (changes requested):
A few things to fix here. Details in comments

Drewienkopl committed:
feat: fixing task 3 stuff pointed out in the comments

I am pretty sure that this is everything mentioned in comments fixed, now everything should be alright.

Drewienkopl committed:
Merge branch 'hubert-drewlo' into hubert-drewlo-3

Drewienkopl committed:
fix: delete old img and prettier

Drewienkopl committed:
fix: prettier2
